src/game/objects/har.c

```diff
@@ -1630,14 +1630,14 @@ void har_collide(object *obj_a, object *obj_b) {
 static void process_range(const har *h, damage_tracker *damage, vga_palette *pal, uint8_t step) {
     // For player 0, we should use palette indexes 1 through 47. For player 1, 49 through 95 (skip black).
     // If pe flag is on, we need to switch to handling the other HAR.
-    const int start = 48 * (h->player_id ^ h->p_har_switch) + 1;
-    const int end = start + 47;
+    const int first = 48 * (h->player_id ^ h->p_har_switch) + 1;
+    const int last = first + 46;
```

Comment:
47

Reply:
it's 46, because we're setting one of two ranges:
1 thru 47, or 49 thru 95.

47 - 1 and 95 - 49 are both 46.
I can write it as 47 - 1, if that helps

Reply:
Both HARs have 48 colors. First color is black, 47 are palette.

Reply:
I find these range changes confusing. Previously it was much easier to reason around the ranges -- now we seem to need -1 offset everywhere.

Reply:
Rewrote the fix, so now damage_add_range takes an exclusive range instead of an inclusive range.